src/repositories/__tests__/request-repository.test.ts

```diff
@@ -131,4 +167,221 @@ describe('request repository test', () => {
+
+      const updatedRequests = await requestRepository.findAndMarkReady(streamLimit)
+      expect(updatedRequests.length).toEqual(streamLimit)
+
+      // get earliest 4 pending as the expired processing request should be the first one
+      const earliestPendingRequestCids = createdRequests
+        .filter(({ status }) => status === RequestStatus.PENDING)
+        .slice(0, streamLimit - 1)
+        .map(({ cid }) => cid)
+
+      expect(updatedRequests.map(({ cid }) => cid)).toEqual([
+        expiredProcessing[0].cid,
+        ...earliestPendingRequestCids,
+      ])
+    })
+
+    test('Marks requests for same streams as ready', async () => {
+      const streamLimit = 5
+      const repeatedStreamId = new StreamID('tile', await randomCID()).toString()
+      const requests = await Promise.all([
+        generateRequests(
+          {
+            status: RequestStatus.PROCESSING,
+            createdAt: new Date(Date.now() - MS_IN_HOUR * 24),
+            updatedAt: new Date(Date.now() - MS_IN_HOUR * 7),
+            streamId: repeatedStreamId,
+          },
+          1
+        ),
+        generateRequests(
+          {
+            status: RequestStatus.PENDING,
+            streamId: repeatedStreamId,
+          },
+          1
+        ),
+        generateRequests(
+          {
+            status: RequestStatus.PENDING,
+          },
+          streamLimit
+        ),
+      ]).then((arr) => arr.flat())
+      const expiredProcessingRequest = requests[0]
+      const reRequested = requests[1]
+
+      const requestRepository = container.resolve<RequestRepository>('requestRepository')
+      await requestRepository.createRequests(requests)
+
+      const createdRequest = await getAllRequests(connection)
+      expect(createdRequest.length).toEqual(requests.length)
+
+      const updatedRequests = await requestRepository.findAndMarkReady(streamLimit)
+      expect(updatedRequests.length).toEqual(streamLimit + 1)
+
+      const updatedRequestCids = updatedRequests.map(({ cid }) => cid)
+      expect(updatedRequestCids).toContain(expiredProcessingRequest.cid)
+      expect(updatedRequestCids).toContain(reRequested.cid)
+    })
+
+    test('Does not mark any transaction as ready if an error occurs', async () => {
+      const streamLimit = 5
+      const requests = await generateRequests(
+        {
+          status: RequestStatus.PENDING,
+        },
+        streamLimit
+      )
+
+      const requestRepository = container.resolve<RequestRepository>('requestRepository')
+      await requestRepository.createRequests(requests)
+
+      const originaUpdateRequest = requestRepository.updateRequests
+      requestRepository.updateRequests = (fields, requests, manager) => {
+        throw new Error('test error')
+      }
+
+      try {
+        await requestRepository.findAndMarkReady(streamLimit).catch(() => {})
+        const requestsAfterUpdate = await getAllRequests(connection)
```

Comment:
I have changed things to "expect that this function will throw". I don't think the other checks should be in the `finally` because the `finally` is just for cleanup.

Reply:
Oh, I had totally missed the "catch" chained onto the promise in the original version.  But yeah, explicitly expecting it to throw is even better